### DISCUSSION TITLE
Can't get journal to work outside of demo

### DISCUSSION THREAD

**Discussion by sonofevil** [TriliumNext/Trilium#6333] [Q&A] [answered]
Alt+t creates a day note inside the Journal demo.

I can't get that functionality to work anywhere else and don't understand why.

If I duplicate the demo journal, the functionality isn't retained. Alt+t still only creates a day note inside the demo.

The duplicated Journal has no external relations, all its relations are to its own children, and have the same structure as the demo, so that can't be the reason. The "Today" script also does not seem to reference anything outside. Structurally and otherwise the duplicate journal is exactly the same as the demo journal. Yet it does not do the same thing.

What's extra weird is that if I change the keyboard shortcut inside the scripts, it has absolutely no effect. If I change it in the duplicate it doesn't work. If I change it in the demo from alt+t to something else, alt+t still continues to work as before.

None of this makes sense to me.

**Comment by zadam** [comment, marked as answer]
Hi, so I think it might be because there are now 2 journals. When you hit Today button it will just find the first one (original) and create the today note there. This whole journal functionality works with assumption that there is only one journal tree, otherwise it might behave like this I guess.

So my advice would be to simply delete the demo journal (you might also want to restart the app to make sure the correct Today button is loaded)

**Comment by sonofevil** [comment]
You are correct, thx. It should be noted that apparently the first Journal is selected by order of creation, with the order inside the tree playing no role.

I think it would be great if we could simultaneously have multiple journals with hotkey functionality.

From what I can gather it seems all the script does is run at launch, creating a toolbar button with an associated keyboard shortcut. The button is then anchored to the journal that created it.

So I guess it might be possible to create multiple buttons for separate journals. Not quite ideal for my use case, it would be better if Trilium created a day note inside the note I've currently selected. Then I can easily have an arbitrary number of Journals.

**Comment by sonofevil** [comment]
I tried replacing `api.getTodayNote()` in the script with `api.getActiveNote()` but that just makes it stop working.
